File: core/src/io/cucumber/factory/BonusFactory.java
package io.cucumber.factory;

import com.badlogic.gdx.graphics.Texture;
import io.cucumber.model.Bonus;

import static io.cucumber.constant.GameConstants.BONUS_LIFESPAN;
import static io.cucumber.constant.GameConstants.BONUS_SIZE;

public class BonusFactory {

    public static Bonus create(float x, float y, Texture texture) {
        return new Bonus(x, y, BONUS_SIZE, texture, BONUS_LIFESPAN);
    }

    private BonusFactory() {
    }
}

File: core/src/io/cucumber/factory/BonusFactory.kt
package io.cucumber.factory

import com.badlogic.gdx.graphics.Texture
import io.cucumber.constant.BonusConstants.BONUS_LIFESPAN
import io.cucumber.constant.BonusConstants.BONUS_SIZE
import io.cucumber.model.Bonus

object BonusFactory {

    fun create(x: Float, y: Float, texture: Texture): Bonus = Bonus(x, y, BONUS_SIZE, texture, BONUS_LIFESPAN)

}